src/deadline/nuke_submitter/deadline_submitter_for_nuke.py

```diff
@@ -78,7 +79,70 @@ def _handle_step(step: dict):
         _handle_step(step)
 
 
-def _get_job_template(settings: RenderSubmitterUISettings) -> dict[str, Any]:
+def _add_gizmos_to_job_bundle(job_bundle_path: Path, nodes: list[nuke.Node]) -> Optional[Path]:
+    gizmo_paths = [Path(node.filename()) for node in nodes if isinstance(node, nuke.Gizmo)]
+    if not gizmo_paths:
+        return None
+    gizmo_dir = job_bundle_path / "gizmos"
+    gizmo_dir.mkdir(parents=True, exist_ok=True)
+    for gizmo_path in gizmo_paths:
+        shutil.copy(gizmo_path, gizmo_dir / gizmo_path.name)
+    return gizmo_dir
+
+
+def _add_gizmo_dir_to_job_template(job_template: dict[str, Any], relative_gizmo_dir: Path) -> None:
+    if "parameterDefinitions" not in job_template:
+        job_template["parameterDefinitions"] = []
+
+    job_template["parameterDefinitions"].append(
+        {
+            "name": "GizmoDir",
+            "type": "PATH",
+            "description": "The directory containing Nuke Gizmo files used by this job.",
+            "default": str(relative_gizmo_dir),
+            "objectType": "DIRECTORY",
+            "dataFlow": "IN",
+            "userInterface": {"control": "CHOOSE_DIRECTORY", "label": "Gizmo Directory"},
+        }
+    )
+
+    render_step = None
+    for step in job_template["steps"]:
+        if step["name"] == "Render":
+            render_step = step
+            break
+    if render_step is None:
+        raise RuntimeError("No 'Render' step found in the job template")
+
+    if "stepEnvironments" not in render_step:
+        render_step["stepEnvironments"] = []
+
+    # This needs to be prepended rather than appended
+    # as it must run before the "Nuke" environment.
+    render_step["stepEnvironments"].insert(
+        0,
+        {
+            "name": "AddGizmoPathsToNukePath",
```

Comment:
nit: this will be read by users, not necessarily developers. So we should add spaces here and a description that will show up in the service:

```
Add Gizmos to NUKE_PATH
```
or something like that